### DISCUSSION TITLE
Maximum devices supported in practice

### DISCUSSION THREAD

**Discussion by aik** [Koenkk/zigbee2mqtt#22280] [General]
I tried CC2531 (no good, ok) and then SONOFF Zigbee ZBDongle-P and -E versions. -P was about the same as CC2531 (losing connections, problems with pairing, even with a larger antenna, which was surprising), -E seems to be better.

But with both -P and -E there was a problem when the number of devices in the network reaches about 50 - random devices start failing - they disappear for some period of time or entirely, often need to re-pair. The network is 10+ bulbs, 10 power points, 10+ door sensors, 5 motion sensors, ~10 switches, Ikea/Xiaomi/Tuya*/etc. The coordinator was connected to OrangePi and now it is intel m3 laptop (no difference). A house on a 900m2 block so really not much radio interference.

So when I put all my 58 devices on the same network - problems, devices randomly disappear. Move 10 of them to the older CC2531 - both networks work stable (so interference is really not a problem). Right now 17 devices connected directly to -E which is far below the official limit of 50. And I have bulbs and power points which are allegedly routers.

I always thought one better network should easily do 250-ish devices but not in my case. What do I miss? Thanks,

**Comment by kevincaradant** [comment]
Hi

Based on my experience (currently using Sonoff P with 100 devices, everything works well), it seems like your issue can come from:

```
1) I suggest checking with the "WiFi Analyzer" app to ensure that your Wi-Fi channel and neighboring Wi-Fi channels do not overlap with your Zigbee channel.

2)Place your coordinator on USB 2 instead of USB 3. Use a USB extension of around 1m to 2m max to distance it from your electrical installations (internet box, computer, SSD, etc.).

3) Review the device connections to your router or coordinator using the schema generated via the Z2M UI. It's possible that your devices are connected to a router that loses connection, or there might be incompatible end devices paired with the router.
cf attachment

4) Low battery levels in end devices often lead to broken links in the generated schema. This could be an initial clue that the battery is starting to run low. Eventually, the device may drop out of the network altogether.

5) Check the logs for any devices that may be spamming the network or have a poor signal.

6) Consider updating to the latest firmware version available from Koenkk's GitHub repository (20230507 from my memory).
```

I hope these suggestions help you resolve your issue.

attachment: 
![image](https://github.com/Koenkk/zigbee2mqtt/assets/6093175/86fcb56e-28d0-45c2-9ef6-78d057bf2384)

**Reply by pvvx** [reply]
> I always thought one better network should easily do 250-ish devices but not in my case.

The limitation of Zigbee depends on the congestion of the radio channel and the speed of the communication interface with the coordinator.
If on average data from 250 devices arrives every 10 seconds, then the RF channel bandwidth will no longer be enough.
But there may be a large number of end devices that communicate infrequently or only when an event occurs, such as switches.

Typical Poll Control 4.7 seconds (zigbee polling interval)
"10 power points" have a data transmission interval of 5 seconds. This is already 2 transactions per second.
In a small Zigbee test network of 11 different devices, 5..10 packets are transmitted per second.

Zigbee network has high latency and is used for devices with long polling intervals, which lags behind the needs of modern Internet of Things.

**Reply by ccaarrsstteenn** [reply]
I have the same problems!
I have a very small Zigbee network in my Loggia and a bigger (43 devices @ Home)
The WIFI channel is 6 and the Zigbee channel is 11 (no other neighbours in the forrest)
<img width="718" alt="Bildschirmfoto 2024-05-03 um 09 48 08" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/18a83fed-5aaa-4ed4-8c1b-df55e570aeab">
the motion Sensor is marked as offline
the Haustuer contact was triggered minutes before.
**why ist it no longer connected?**

The battery is more than 80% for all devices.
The log is looking stable.
Can I create a logfile over days, to look up, when the sensor "left" the network?
Can I use another coordinator? Which is the best? I am using zStack3x0 from Sonoff Coordinator-Version 20230507

In my scenario one by one device is apsent in the network. Sometimes no longer in the list (as the device has left the network), sometimes only offline. Then I configure the Zigbee Network to allow devices. Reset The divice (long Press at the device) and the device is joining the network, battery 80%, working fine for the nest 1,2,3,4 weeks, then the same with the same device or another device.

Can anybody help to solve this problem?

**Reply by kevincaradant** [reply]
The % sometimes is not an exact science ( aqara for example). Sometimes the device will die at 100%. 
Yesterday. 100% displayed at home 2985mv.
Device down....
C ant pair it again, the blue light is low and bliking.
I changed the battery. Works again with a pretty blue light.

I used two coordinators for 2 different setup on different location. Sonoff Dongle P and LAUNCHXL-CC1352P-2.

Since 6 months for Sonoff and 4 years for CC 1352P2 everything seems rock and work well

I Just watched this early.
https://youtu.be/yY-aD1hvwr4

Maybe it could be a solution ?

**Comment by ccaarrsstteenn** [comment]
Here are some additional information:
<img width="1964" alt="Bildschirmfoto 2024-05-05 um 12 13 44" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/c3dba7fe-7ce7-4ea6-baf1-7513dfb9fced">
as you can see in the first diagram the device is leaving ...
after pressing the button on the device:
<img width="647" alt="Bildschirmfoto 2024-05-05 um 12 27 28" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/acb7cb71-f193-408c-8728-ef280e24a469">
everything is looking fine, but the values are from 8 hours ago
<img width="517" alt="Bildschirmfoto 2024-05-05 um 12 50 35" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/ba22ef6a-a12f-4144-aad7-b8b2d230fa84">
<img width="2213" alt="Bildschirmfoto 2024-05-05 um 12 51 03" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/c863b2fa-42f8-4fe9-8c3f-758559559838">
last seen is from 1 minute ago.
<img width="564" alt="Bildschirmfoto 2024-05-05 um 12 54 57" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/5d4f4b4d-a001-4cb1-9d81-4562e255603e">
a little bit later snapshor from dashboard, but with old values.
The log on debug level:
<img width="1428" alt="Bildschirmfoto 2024-05-05 um 12 58 20" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/4794b146-c8a2-4d81-9466-26a3b7d0c9c6">

**Comment by ccaarrsstteenn** [comment]
I forgot the binding settings for this device:
<img width="2060" alt="Bildschirmfoto 2024-05-05 um 12 50 52" src="https://github.com/Koenkk/zigbee2mqtt/assets/120797888/c1cc4d7e-434d-4fe7-a53e-1bbfda1bdf3c">
